crates/cli/src/subcommands/generate/csharp.rs

```diff
@@ -407,28 +407,28 @@ fn autogen_csharp_access_funcs_for_struct(
             None => continue,
             Some(x) => x,
         };
-        writeln!(
-            output,
-            "public readonly ref struct {csharp_field_name_pascal}UniqueIndex"
-        );
+        writeln!(output, "public class {csharp_field_name_pascal}UniqueIndex");
         indented_block(output, |output| {
+            write!(
+                output,
+                "internal readonly Dictionary<{csharp_field_type}, {csharp_table_name}> __Cache = new(16);"
+            );
+            writeln!(output);
+
             writeln!(
                 output,
                 "public {struct_name_pascal_case}? Find({csharp_field_type} value)"
```

Comment:
If we're using a class, might as well extract those into a shared base class, but should be good to go as-is for now too.